games/maze/src/maze.rs

```diff
@@ -115,6 +131,15 @@ impl Maze {
         Ok(())
     }
 
+    pub fn give_player_directions(&self) -> GiveDirections {
+        GiveDirections {
+            left: self.get_tile_type(self.player.x as i32 - 1, self.player.y as i32),
+            right: self.get_tile_type(self.player.x as i32 + 1, self.player.y as i32),
+            up: self.get_tile_type(self.player.x as i32, self.player.y as i32 - 1),
+            down: self.get_tile_type(self.player.x as i32, self.player.y as i32 + 1),
+        }
+    }
```

Comment:
`self.player.x as i32` and `self.player.y as i32` are fairly noisy here, might be better to put them as temp variables above:
```rust
let x = self.player.x as i32;
let y = self.player.y as i32;
...
```